rideshare.rb

```diff
@@ -0,0 +1,141 @@
+all_rides = [
+  {
+    "driver id" => "DR0001",
+    "rides" => [
+      {
+        "date" => "3rd Feb 2016",
+        "cost" => 10,
+        "rider_ID" => "RD0003",
+        "rating" => 3,
+      },
+      {
+        "date" => "3rd Feb 2016",
+        "cost" => 30,
+        "rider_ID" => "RD0015",
+        "rating" => 4,
+      },
+      {
+        "date" => "5th Feb 2016",
+        "cost" => 45,
+        "rider_ID" => "RD0003",
+        "rating" => 2,
+      },
+    ],
+  },
+  {
+    "driver id" => "DR0002",
+    "rides" => [
+      {
+        "date" => "3rd Feb 2016",
+        "cost" => 25,
+        "rider_ID" => "RD0073",
+        "rating" => 5,
+      },
+      {
+        "date" => "4th Feb 2016",
+        "cost" => 15,
+        "rider_ID" => "4th Feb 2016",
+        "rating" => 1,
+      },
+      {
+        "date" => "5th Feb 2016",
+        "cost" => 35,
+        "rider_ID" => "RD0066",
+        "rating" => 3,
+      },
+    ],
+  },
+  {
+    "driver id" => "DR0003",
+    "rides" => [
+      {
+        "date" => "4th Feb 2016",
+        "cost" => 5,
+        "rider_ID" => "RD0066",
+        "rating" => 5,
+      },
+      {
+        "date" => "5th Feb 2016",
+        "cost" => 50,
+        "rider_ID" => "RD0003",
+        "rating" => 2,
+      },
+    ],
+  },
+  {
+    "driver id" => "DR0004",
+    "rides" => [
+      {
+        "date" => "3rd Feb 2016",
+        "cost" => 5,
+        "rider_ID" => "RD0022",
+        "rating" => 5,
+      },
+      {
+        "date" => "4th Feb 2016",
+        "cost" => 10,
+        "rider_ID" => "RD0022",
+        "rating" => 4,
+      },
+      {
+        "date" => "5th Feb 2016",
+        "cost" => 20,
+        "rider_ID" => "RD0073",
+        "rating" => 5,
+      },
+    ],
+  },
+]
+
+driver_data = [] # This array will store hashes that will store data throughout the upcoming calculations.
+all_rides.each do |d|
+  driver = {} # Stores each driver's stats
+  driver["driver id"] = d["driver id"]
```

Comment:
Similar to the feedback I left on Portmanteau generator, this code would be much easier to read if you were to break it up using blank lines. Not sure if that feedback came in time to help here, but I'll be watching for this on Adagrams.